drivers/hmis/spec/requests/hmis/client_search_spec.rb

```diff
@@ -120,60 +120,79 @@
-      # OTHER FILTERS
-      ['personal id', { personal_id: 'db422f5fff0b8f1c9a4b81f01b00fdb4' }, true],
-      ['wrong personal id and not match', { personal_id: '00000000000000000000000000000000' }, false],
-      ['warehouse id', { warehouse_id: '5555' }, true],
-      ['wrong warehouse id', { warehouse_id: '5556' }, false],
-      ['first name', { first_name: 'William' }, true],
-      # TODO: Test nickname match
-      # TODO: Test metaphone match
-      ['wrong first name and not match', { first_name: 'Dave' }, false],
-      ['last name', { last_name: 'Smith' }, true],
-      # TODO: Test nickname match
-      # TODO: Test metaphone match
-      ['wrong last name and not match', { last_name: 'Jones' }, false],
-      ['last 4 of ssn', { ssn_serial: '6789' }, true],
-      ['wrong last 4 of ssn and not match', { ssn_serial: '0000' }, false],
-      ['dob d/m/yyyy', { dob: '1/12/1999' }, true],
-      ['dob yyyy-mm-dd', { dob: '1999-12-01' }, true],
-      ['wrong dob and not match', { dob: '2000-12-01' }, false],
-      # TODO: Projects filter
-      # TODO: Organizations filter
-    ].each do |desc, input, match|
-      it "should search by #{desc}" do
+    it 'should search' do
+      [
+        # TEXT SEARCHES
+        *[
+          ['text: first name', 'william', true],
+          ['text: last name', 'smith', true],
+          ['text: last, first', 'smith, william', true],
+          ['text: first last', 'william smith', true],
+          ['text: partial names', 'wi sm', true],
+          ['text: wrong name and not match', 'x x', false],
+          ['text: personal id', 'db422f5fff0b8f1c9a4b81f01b00fdb4', true],
+          ['text: wrong personal id and not match', '00000000000000000000000000000000', false],
+          ['text: ssn', '123-45-6789', true],
+          ['text: wrong ssn and not match', '000-00-0000', false],
+          ['text: dob', '12/01/1999', true],
+          ['text: wrong dob and not match', '12/01/2000', false],
+          ['text: warehouse id', '5555', true],
+          ['text: scan card code', 'P1234', true],
+          ['text: deactivated scan card code', 'P5678', false],
+          ['text: expired scan card code', 'P6666', false],
+          ['text: scan card code for another client', 'P9999', false],
+          ['text: enrollment id', "enrollment:#{e1.id}", true],
+          ['text: WIP enrollment id', "enrollment:#{e2_wip.id}", true],
+          ['text: enrollment id without correct prefix', e1.id, false],
+          ['text: wrong enrollment id', 'enrollment:123', false],
+          ['text: wrong enrollment id (other hhm\'s id)', "enrollment:#{e3_other_hhm.id}", false],
+          ['text: household id', "household:#{e1.household_id}", true],
+          ['text: numeric household id', 'household:999', true],
+          ['text: WIP household id', "household:#{e2_wip.household_id}", true],
+          ['text: household id without correct prefix', e1.household_id, false],
```

Comment:
Moved these guys into a single `it` so that we could reference variables here, which feels very necessary. Is there a better way to handle that?

I added custom error messages so they still show you which case failed.

Reply:
seems reasonable.